fury/ui/helpers.py

```diff
@@ -44,14 +44,91 @@ def clip_overflow(textblock, width, side='right'):
     while start_ptr < end_ptr:
         mid_ptr = (start_ptr + end_ptr)//2
         textblock.message = original_str[:mid_ptr] + "..."
-        if textblock.size[0] < width:
-            start_ptr = mid_ptr
-        elif textblock.size[0] > width:
-            end_ptr = mid_ptr
 
-        if mid_ptr == (start_ptr + end_ptr)//2 or\
-           textblock.size[0] == width:
+        is_overflowing, start_ptr, end_ptr =\
+            check_overflow(textblock, width, start_ptr, end_ptr)
+
+        if not is_overflowing:
             textblock.have_bg = prev_bg
             if side == 'left':
                 textblock.message = textblock.message[::-1]
             return textblock.message
+
+
+def wrap_overflow(textblock, wrap_width):
+    """Wraps overflowing text of TextBlock2D with respect to width.
+
+    Parameters
+    ----------
+    textblock : TextBlock2D
+        The textblock object whose text needs to be wrapped.
+    wrap_width : int
+        Required width of the wrapped text.
+
+    Returns
+    -------
+    wrapped text : str
+        Wrapped version of the text.
+    """
+    original_str = textblock.message
+    start_ptr = 0
+    end_ptr = len(original_str)
+    prev_bg = textblock.have_bg
+    textblock.have_bg = False
+
+    if wrap_width <= 0:
+        wrap_width = textblock.size[0] - abs(wrap_width) % textblock.size[0]
+
+    if textblock.size[0] <= wrap_width:
+        textblock.have_bg = prev_bg
+        return original_str
+
+    while start_ptr < end_ptr:
+        mid_ptr = (start_ptr + end_ptr)//2
+        textblock.message = original_str[:mid_ptr]
```

Comment:
This entire block can be reused.

Reply:
Hey @Nibba2018, this has been fixed. Thanks!